stdlib/Unicode/src/Unicode.jl

```diff
@@ -89,4 +89,44 @@ letter combined with an accent mark is a single grapheme.)
 """
 graphemes(s::AbstractString) = Base.Unicode.GraphemeIterator{typeof(s)}(s)
 
+const EMOJI_RANGES = [
+    0x1F600:0x1F64F,  # Emoticons
+    0x1F300:0x1F5FF,  # Misc Symbols and Pictographs
+    0x1F680:0x1F6FF,  # Transport and Map
+    0x2600:0x26FF,    # Misc symbols
+    0x2700:0x27BF,    # Dingbats
+    0xFE00:0xFE0F,    # Variation Selectors
+    0x1F900:0x1F9FF,   # Supplemental Symbols and Pictographs
+];
+
+const ZWJ = '\u200d'    # Zero-width joiner
+
+"""
+    isemoji(Union{AbstractChar, AbstractString}) -> Bool
+
+Test whether a character is an emoji, or whether all elements in a given string are emoji. Includes identifying composite emoji.
+"""
+function isemoji(c::AbstractChar)
+    u = UInt32(c)
+    @inbounds for emojiset in EMOJI_RANGES
+        u in emojiset && return true
+    end
+    return false
+end
+
+function isemoji(s::AbstractString)
```

Comment:
```suggestion
function isemoji(s::AbstractString)
    isempty(s) && return true
```
It's a bit unclear what to do if the string is empty: on the one hand, it contains no emoji; on the other hand it contains no characters that aren't emoji. Either way, the empty string needs to be checked for before you can do `s[end]`.

Reply:
Thanks for the catch! This is tricky. `isascii` was my mold for this, and that returns true when an empty string is passed. However, it does feel a bit weird to return true. I think the best thing is to update the docstring to make this behavior clear

Reply:
I agree that `true` is the correct answer here.